skbio/tree/_tree.py

```diff
@@ -3101,7 +3101,7 @@ def shuffle(self, k=None, names=None, shuffle_f=np.random.shuffle, n=1):
 
     @experimental(as_of="0.5.2-dev")
     def support(self):
-        """Return support value of a node if available
+        """Return support value of a node if available.
```

Comment:
Jamie recommended `support` to be a property. Daniel and I didn't like this idea. You can see our conversation in the last PR.

Reply:
I thought it did not make sense to be part of the parser. I thought that it *did* make sense to be a property.

Reply:
@wasade I see. Sorry I didn't fully understand your point. My own preference was to make it a method rather than a hard-coded attribute that every `TreeNode` object has. Here is my [rationale](https://github.com/biocore/scikit-bio/pull/1568#issuecomment-385815407). However, after 2nd thinking I am hesitated.

I foresee that there are many changes that the user wants to add custom attributes to nodes. `support` is likely one of the most-wanted attribute. In such case, the user can do
```
for node in tree.non_tips():
    node.support = node.support()
```
I am not sure if this will work. Even if so, it may be confusing.

How about this: let's make `support` an attribute. But by default (`TreeNode.read`) it will not be parsed. Meanwhile, we create a function `assign_supports`. It will traverse the whole tree and assign value to the `support` attribute. Similar to the existing `assign_ids` function. See [here](http://scikit-bio.org/docs/0.5.2/generated/skbio.tree.TreeNode.assign_ids.html#skbio.tree.TreeNode.assign_ids)

It will also have an extra function: In cases where the `name` of an internal node is like "80:genusA", the `assign_supports` will set `support` to be 80 and change `name` into `genusA`.

What do you think? @wasade @RNAer @mortonjt ?

Reply:
that's not how it is supposed to be implemented. search for `@property`. I think there are even examples in the scikit-bio codebase.

Reply:
For this `support` thing, I searched a little bit and realized that it is way more complicated than I thought. See [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5435079/). It needs to be carefully engineered. How about let's merge this for now, and I will tackle this in the next PR.

Reply:
ok!